File: core/common/src/main/java/ru/korus/tmis/lis/data/MicroOrganismResult.java
package ru.korus.tmis.lis.data;

import javax.xml.bind.annotation.XmlElement;

/**
 * Результат анализа микроорганизмов
 */

public class MicroOrganismResult {
    /**
     * идентификатор организма по БД ЛИС
     */
    private String organismCode;

    /**
     * название организма
     */
    private String organismName;

    /**
     * концентрация организма
     */
    private String organismConcentration;

    @XmlElement(name = "organismLisId")
    public String getOrganismCode() {
        return organismCode;
    }

    public void setOrganismCode(String organismCode) {
        this.organismCode = organismCode;
    }

    @XmlElement(name = "organismName")
    public String getOrganismName() {
        return organismName;
    }

    public void setOrganismName(String organismName) {
        this.organismName = organismName;
    }

    @XmlElement(name = "organismConcentration")
    public String getOrganismConcentration() {
        return organismConcentration;
    }

    public void setOrganismConcentration(String organismConcentration) {
        this.organismConcentration = organismConcentration;
    }
}

File: core/business-tier/src/main/scala/ru/korus/tmis/laboratory/altey/accept/MicroOrganismResult.scala
package ru.korus.tmis.laboratory.altey.accept

import ru.korus.tmis.laboratory.altey.{accept2 => lab}
import scala.language.implicitConversions

final object MicroOrganismResult {
  implicit def fromLab1(v: ru.korus.tmis.laboratory.altey.accept2.MicroOrganismResult): MicroOrganismResult = {
    MicroOrganismResult(Option(v.getOrganismCode), Option(v.getOrganismName), Option(v.getOrganismConcentration))
  }
}

/**
 * Результат анализа микроорганизмов
 */
final case class MicroOrganismResult(
                                      /**
                                       * идентификатор организма по БД ЛИС
                                       */
                                      organismCode: Option[String],

                                      /**
                                       * название организма
                                       */
                                      organismName: Option[String],

                                      /**
                                       * концентрация организма
                                       */
                                      organismConcentration: Option[String]
                                      )

